### PR TITLE
Updating markup to reflect new linting rules

### PR DESCRIPTION
**Pull request by ryelo**
This changes markup, but is non breaking

## Curly Braces
These will apply to props and arrays inside of arrays.  Before, these only worked on the "top most level" (1) and wouldn't check if the curly braces were inside another array or inside of an object.

(1) Edit: Top level meaning everything outside of return statements

The only thing I'm still investigating is, for example, in RulesCard where we're setting HTML inside. I might just rewrite it and sanitize it, then use it as a variable. Ideally, if you need to use double curly braces, IMO it should be like

 `{{ foo }}`  instead of `{ { foo } }`

However, I don't hate the second way.

It might just need to a contribution to the React eslint plugin because I can't find anything about it anywhere which might end up with me opening a PR.

## Brackets
Brackets now follow similar rules to curly braces where they require a space inside.
Formerly: `[foo, bar]`
Proposed: `[ foo, bar ]`

### CODE REVIEW

File: .eslintrc.yml [hunk 1]
```diff
@@ -83,6 +83,7 @@ rules:
     - error
     - always
     - objectsInObjects: false
+      arraysInObjects: false
```
Comment (AllenBW) [inline .eslintrc.yml]: should read `- arraysInObjects: false`?

File: .eslintrc.yml [hunk 2]
```diff
@@ -71,6 +82,8 @@ rules:
   object-curly-spacing:
     - error
     - always
+    - objectsInObjects: false
+      arraysInObjects: false
```
Comment (AllenBW) [inline .eslintrc.yml]: should read `- arraysInObjects: false` ?

**Comment by AllenBW** [comment]
> {{ foo }} instead of { { foo } }

I hate that second way 😏 🤗

File: .eslintrc.yml (hunk 2 above)
Comment (ryelo) [reply .eslintrc.yml]: It should be like this because these exceptions are read as one attribute, even if there are two of them.

Comment (ryelo) [reply .eslintrc.yml]: Example: https://github.com/RedHatInsights/insights-advisor-frontend/blob/506bd384c4bc1902113de8b4445cf66620776b9f/.eslintrc.yml#L98

**Comment by AllenBW** [comment]
Also, ps, who is deciding how or what we style? IMO we ought to just be done with it, (the act of styling and wiggling back n fourth on it) and just do what the pfr repo is doing. The configuration [here](https://github.com/patternfly/patternfly-react/blob/master/.prettierrc) and [here](https://github.com/patternfly/patternfly-react/blob/master/.stylelintrc) is **far** less than whats going on in this repo (yeah this might be something to bring up in the starter app also

**Comment by ryelo** [comment]
@AllenBW Partially that is because they're being consumed by an NPM package and not being deployed as an app - especially the starter app. There are ~7-8 apps that, in the end, will be going to consuming the starter app, if we have consistent styling across all of them, it'll be easy for a person to be able to contribute anywhere and know the code standards.

**Comment by AllenBW** [comment]
Cool so then why not get behind what pfr is doing rather considering we do/will consume a great deal of it?  It really boils down to avoiding prs like this (and all the future ones), [cuz these deps already have it figured out...](https://github.com/patternfly/patternfly-react/blob/master/package.json#L56-L60)

**Comment by ryelo** [comment]
They're importing [airbnb's style](https://github.com/airbnb/javascript/tree/master/packages/eslint-config-airbnb-base/rules) which is *much* more extensive than ours is.

Again, people consume pf as components through an npm package, not through a repo which they have to build on top of and other devs have to support.

**Comment by ryelo** [comment]
@AllenBW we don't have to merge this. If we want to import another linter like airbnb's, that's fine. But as of right now, there are multiple places that break rules in this app. The only requirement is that if something breaks rule, it needs to throw an error and the app needs to break so that it doesn't hit the webhook that deploys it.

**Comment by AllenBW** [comment]
This is the fastest way to whole @ryelo, am totally down to merge. Bringing up the discussion [over here](https://github.com/RedHatInsights/insights-frontend-starter-app/issues/46) cuz that's where it really belongs.

**Comment by ryelo** [comment]
@AllenBW Mind rechecking the files you changes to make sure I hit everything and it's working correctly?

**Comment by AllenBW** [comment]
jumping in, please stand by...

**Comment by AllenBW** [comment]
The only thing is.... `Warning: React version not specified in eslint-plugin-react settings. See https://github.com/yannickcr/eslint-plugin-react#configuration. `

`.eslintrc.yml`
```
settings:
  react:
    version: '16.0'
```

**Comment by ryelo** [comment]
@AllenBW updated!

**Pull request merged**